Autopilot/Test/Src/TelemetryManager/Test_TelemetryFSM.cpp

```diff
@@ -0,0 +1,110 @@
+#include <gtest/gtest.h>
+
+#include "telemetryManager.hpp"
+#include "telemetryStateClasses.hpp"
+
+
+using namespace std; 
+using ::testing::Test;
+
+TEST(TelemetryManagerFSM, InitialStateIsInitialMode){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), initialMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, InitialTransitionToGround){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), groundDataPathMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToRegularThroughDataValid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDECIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), regularReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToRegularThroughDataInvalid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), regularReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToErrorThroughDataValid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=true;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();
+    telemetryMng.execute();
+    /***ASSERTS***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), errorReportMode::getInstance());
+}
+
+TEST(TelemetryManagerFSM, GroundToErrorThroughDataInvalid){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute(); //to ground
+    for(int index=0; index<4; index++)
+    {
+        telemetryMng.execute();//to regular report
+        telemetryMng.execute();//to ground
+    }
+    //last increase in counter
+    telemetryMng.execute();//to error report
+    /***ASSERT***/
+    ASSERT_EQ(*(telemetryMng.getCurrentState()), errorReportMode::getInstance());
+}
+TEST(TelemetryManagerFSM, RegularToGround){
+    /***SETUP***/
+    telemetryManager telemetryMng;
+    /***DEPENDENCIES***/
+    telemetryMng.dataValid=true;
+    telemetryMng.dataError=false;
+    /***STEPTHROUGH***/
+    telemetryMng.execute();//to ground 
+    telemetryMng.execute();//to regular report
```

Comment:
I see what you are trying to do here. Instead of writing these two lines, just write: 

`telemetryMng.setState(regularReportMode::getInstance());`

That way your code does not depend on the transition from previous parts. That way if the test fails, you are 100% sure it's because of he regualrReportMode state.

Reply:
got it, will change